### ISSUE TITLE
contrib/internal/httputil: WriteHeader allows writing status code multiple times

### ISSUE THREAD

**Issue by cweerasooriya** [DataDog/dd-trace-go#623]
Log entry displays the wrong status when `ResponseWriter.WriteHeader` is invoked multiple times.

Eg.
```
package main

import (
	"fmt"
	"net/http"
	"gopkg.in/DataDog/dd-trace-go.v1/contrib/gorilla/mux"
)

func main() {
	r := mux.NewRouter()

	r.HandleFunc("/test", func(w http.ResponseWriter, r *http.Request) {
		w.WriteHeader(http.StatusOK)

		w.WriteHeader(http.StatusForbidden) // this call returns early at net/http response.WriteHeader which prevents the request status being updated
	}).Methods("GET")

	http.ListenAndServe(":80", r)
}
```

**Result:**
Request status: 200 OK
`http.status_code` in log: 403 Forbidden

**Some more details:**
Updating HTTP status of a response multiple times within the same handler is prevented by Go by returning early in [response.WriteHeader](https://github.com/golang/go/blob/master/src/net/http/server.go#L1121) function.

The problem seem to be in the DD wrapped [responseWriter](https://github.com/DataDog/dd-trace-go/blob/v1/contrib/internal/httputil/trace.go#L67) where it assigns the response.status without checking whether it's set before.

**Comment by gbbr** [comment]
You're right, this is wrong. I'll add it to our backlog. In the meantime, if you'd like to address it - we gladly welcome PRs.

**Comment by knusbaum** [comment]
This fix is available in `v1.23.1` and newer versions of dd-trace-go.
